concepts/arrays/about.md

```diff
@@ -1 +1,157 @@
+
+Let's walk through an example to illustrate how this works:
+
+```rust
+fn main() {
+    // Assume we have an array-like structure called `a`
+    let mut a = ArrayTrait::new();
+
+    // Add some elements to the array
+    a.append(10);
+    a.append(1);
+    a.append(2);
+
+    // Remove the first element from the front of the array
+    let first_value = a.pop_front().unwrap(); // `unwrap()` is used to get the value from `Option`
+
+    // Print the removed value
+    println!("The first value is {}", first_value);
+}
+```
+
+The `pop_front()` method is used to remove the first element from the front of the array. In this case, `10` is removed.
+
+`pop_front()` returns an `Option`. This allows for handling situations where the array might be empty (we will show how this can be done later in the course).
+
+`unwrap()` is used here assuming the array isn't empty, directly retrieving the value of the removed element (`10`). In real-world scenarios, error handling would typically be more robust.
+
+## Accessing Values in an Array
+
+To access array elements, you can use `get()` or `at()` array methods that return different types. An equivalent to `arr.at(index)` is the subscripting operator `arr[index]`.
+
+### 1. Using get() Method:
+
+The `get()` method is like looking up an item in a list by its number.
+Here’s how it works:
+
+- arr.get(index) allows you to fetch an item at a specific position (index) in the array.
+- It returns an Option, which means:
+  Some(value): If the item exists at index, it gives you the value of that item.
+  None: If the index is out of bounds or doesn't exist in the array.
+- This method is ideal when you want to handle cases where the index might be invalid without causing your program to crash.
+
+```rust
+fn main() {
+    // Create an array to hold unsigned 128-bit integers
+    let mut arr = ArrayTrait::<u128>::new();
+
+    // Add an element to the array
+    arr.append(18);
+
+    // Define the index to access (change this to see different results)
+    let index_to_access = 1;
+
+    // Use match to handle the result of accessing the element at the specified index
+    match arr.get(index_to_access) {
+        Some(value) => {
+            // Print the value if the element exists at the index
+            println!("Element at index {}: {}", index_to_access, *value);
+        },
+        None => {
+            // Panic with a message if the index is out of bounds
+            panic!("Index {} is out of bounds!", index_to_access);
+        }
+    }
+}
+```
+
+### 2. Using at() Method:
+
+The `at()` method directly gives you a snapshot of the element at a specific index using the `unbox()` operation to extract the stored value from the box. If you try to access an index that doesn't exist (out-of-bounds), the program will panic with an error. This method is useful when you want your program to stop if it encounters an unexpected situation like accessing an invalid index.
```

Comment:
```suggestion
The `at()` method, as well as its equivalent the subscriptor operator, directly give you a snapshot of the element at a specific index using the `unbox()` operation to extract the stored value from the box. If you try to access an index that doesn't exist (out-of-bounds), the program will panic with an error. This method is useful when you want your program to stop if it encounters an unexpected situation like accessing an invalid index.
```
Let's merge the `at()` method example with the subscriptor operator example, see https://github.com/cairo-book/cairo-book/pull/861/files#diff-8f38abc2e0eb61af4e7ea66d4dafaf8171c6e801947209b453eb81b677156b64